packages/workflow/src/Extensions/ExpressionExtension.ts

```diff
@@ -74,10 +78,38 @@ export const hasNativeMethod = (method: string): boolean => {
 	});
 };
 
+// /**
+//  * recast's types aren't great and we need to use a lot of anys
+//  */
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function parseWithEsprimaNext(source: string, options?: any): any {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+	const ast = esprimaParse(source, {
+		loc: true,
+		locations: true,
+		comment: true,
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		range: getOption(options, 'range', false),
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		tolerant: getOption(options, 'tolerant', true),
+		tokens: true,
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		jsx: getOption(options, 'jsx', false),
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		sourceType: getOption(options, 'sourceType', 'module'),
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	} as any);
+
+	return ast;
+}
```

Comment:
Minor details: We could type `options` as `esprima.Config` and assert the return types of `getOption` (instead of ESLint exceptions) and return the result directly without assigning.